docs/docs/sourcing-from-yaml.md

```diff
@@ -0,0 +1,232 @@
+        vexillologist post-ironic glossier ennui XOXO mlkshk godard pour-over blog tumblr
+        humblebrag. Blue bottle put a bird on it twee prism biodiesel brooklyn. Blue
+        bottle ennui tbh succulents.
+  links:
+    - to: "/page5"
+    - to: "/page1"
+- path: "/page5"
+  content:
+    - item: St. agur blue cheese queso cheesecake.
+        Cheesecake the big cheese monterey jack cheesecake monterey jack paneer halloumi rubber cheese.
+        Cheese triangles cheese strings cheese slices cheesy feet taleggio cottage cheese when the cheese comes out everybody's happy gouda.
+        Feta cauliflower cheese babybel cheese on toast monterey jack.
+    - item:
+        Bacon ipsum dolor amet short ribs brisket venison rump drumstick pig sausage
+        prosciutto chicken spare ribs salami picanha doner. Kevin capicola sausage,
+        buffalo bresaola venison turkey shoulder picanha ham pork tri-tip meatball meatloaf
+        ribeye. Doner spare ribs andouille bacon sausage. Ground round jerky brisket
+        pastrami shank.
+  links:
+    - to: "/page1"
+```
+
+What the code block above is doing, is nothing more, nothing less, than creating a simple YAML object in which:
+
+- Each `path` is a page's endpoint.
+- The `contents` list holds some data to be displayed.
+- The `links` list holds some endpoints to other pages.
+
+## Gatsby configuration
+
+Now that we have almost all the parts necessary in place, you'll need to make a small adjustment to your Gatsby website.
+
+Start by creating a `gatsby-node.js` with the following contents:
+
+```javascript
+const fs = require("fs")
+const yaml = require("js-yaml")
+exports.createPages = ({ actions }) => {
+  const { createPage } = actions
+  const ymlDoc = yaml.safeLoad(fs.readFileSync("./content/index.yaml", "utf-8"))
+  ymlDoc.forEach(element => {
+    createPage({
+      path: element.path,
+      component: require.resolve("./src/templates/basicTemplate.js"),
+      context: {
+        pageContent: element.content,
+        links: element.links,
+      },
+    })
+  })
+}
+```
+
+Breaking down the code into smaller parts:
+
+1. The `js-yaml` package installed earlier is imported.
+2. The `index.yaml` file is loaded and the content parsed.
+3. Through the use of Gatsby's `createPage()` api, the parsed file will be used to create some pages programmatically.
+4. Using the `context` property, will allow every piece of data you add there, to be accessible in the page as a special prop named `pageContext` allowing it to be consumed. You can read more about it [here](https://www.gatsbyjs.org/docs/creating-and-modifying-pages/)
+
+## Review of the steps so far
+
+Doing a quick review of what you have done so far:
+
+1. Created a new Gatsby website based on a template.
+2. Added some dependencies to handle YAML files.
+3. Created the file and folder structure to house the contents.
+4. Configured Gatsby to fetch the data and create some pages programmatically.
```

Comment:
```suggestion
4. Configured `gatsby-node.js` to fetch the data and create some pages programmatically at build time.
```

Reply:
Can i offer a compromise? 

Why not the following:

> 4. Extended Gatsby's default configuration through `gatsby-node.js` to fetch the data and create some pages programmatically at build time. 



With that small change and focusing on people that stumble onto this document when performing a search on the gatsby website or though a google search will know a priori that Gatsby offers a way to deal with the page creation or some configurations and data management.